File: JS/MiniMaxSum.test.js
/**
 * @version JAVASCRIPT ES6
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ node NomeDaClasse.test
 */
import TEST from 'tape'
import { miniMaxSum } from './MiniMaxSum.js'

TEST('Starting MiniMaxSum test...', (t) => {
    //Test 1
    let want = "10 14"
    let got = miniMaxSum( [1, 2, 3, 4, 5] )
    t.assert(want === got, "Expect: 10 14")
    //Test 2
    want = "16 24"
    got = miniMaxSum( [1, 3, 5, 7, 9] )
    t.assert(want === got, "Expect: 16 24")
    //Test 3
    want = "299 9271"
    got = miniMaxSum( [7, 69, 2, 221, 8974] )
    t.assert(want === got, "Expect: 299 9271")
    //Test 4
    want = "1673711044 2486347135"
    got = miniMaxSum( [140638725, 436257910, 953274816, 734065819, 362748590] )
    t.assert(want === got, "Expect: 1673711044 2486347135")

    t.end()
})

File: JAVA/MiniMaxSumTest.java
/** 
 * @version JAVA
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ mvn clean test -Dtest=your.package.TestClassName
 */ 
package JAVA;

import static org.junit.Assert.assertEquals;

import java.util.ArrayList;
import java.util.List;

import org.junit.Test;

public class MiniMaxSumTest {
    
    @Test
	public void testMiniMaxSum() throws Exception {
		//Test 1
		String want = "10 14";
        List<Integer> ops = new ArrayList<Integer>();
        ops.add(1);
        ops.add(2);
        ops.add(3);
        ops.add(4);
        ops.add(5);
		String got = MiniMaxSum.miniMaxSum( ops );
        assertEquals(want, got);
		//Test 2
		want = "16 24";
        ops = new ArrayList<>();
        ops.add(1);
        ops.add(3);
        ops.add(5);
        ops.add(7);
        ops.add(9);
		got = MiniMaxSum.miniMaxSum( ops );
        assertEquals(want, got);
		//Test 3
		want = "299 9271";
        ops = new ArrayList<>();
        ops.add(7);
        ops.add(69);
        ops.add(2);
        ops.add(221);
        ops.add(8974);
		got = MiniMaxSum.miniMaxSum( ops );
        assertEquals(want, got);
		//Test 4
		want = "1673711044 2486347135";
        ops = new ArrayList<>();
        ops.add(140638725);
        ops.add(436257910);
        ops.add(953274816);
        ops.add(734065819);
        ops.add(362748590);
		got = MiniMaxSum.miniMaxSum( ops );
        assertEquals(want, got);
	}

}
